compiler/rustc_codegen_llvm/src/callee.rs

```diff
@@ -181,7 +181,10 @@ pub fn get_fn(cx: &CodegenCx<'ll, 'tcx>, instance: Instance<'tcx>) -> &'ll Value
                 llvm::LLVMSetDLLStorageClass(llfn, llvm::DLLStorageClass::DllImport);
             }
 
-            if cx.tcx.sess.relocation_model() == RelocModel::Static {
+            let linkage = base::linkage_from_llvm(llvm::LLVMRustGetLinkage(llfn));
+            let visibility = base::visibility_from_llvm(llvm::LLVMRustGetVisibility(llfn));
```

Comment:
At this point consider just passing in the LLVM value to the `should_assume_dso_local` and operate directly with LLVM `Linkage` and `Visibility` types inside there.

Reply:
It would also help to avoid having to figure out a new way to pass in information about thread-locals – can ask this from llval.

Reply:
Addressed these as well now.